src/integrationTest/sam.test.ts

```diff
@@ -53,153 +85,349 @@ async function getSamCodeLenses(documentUri: vscode.Uri): Promise<vscode.CodeLen
             if (!codeLenses || codeLenses.length === 0) {
                 continue
             }
+
             // omnisharp spits out some undefined code lenses for some reason, we filter them because they are
             // not shown to the user and do not affect how our extension is working
-            codeLenses = codeLenses.filter(lens => lens !== undefined && lens.command !== undefined)
-            if (codeLenses.length === 3) {
-                return codeLenses as vscode.CodeLens[]
+            codeLenses = codeLenses.filter(codeLens => {
+                if (codeLens.command && codeLens.command.arguments && codeLens.command.arguments.length === 1) {
+                    // tslint:disable: no-unsafe-any
+                    return (
+                        codeLens.command.command === getInvokeCmdKey(language) &&
+                        codeLens.command.arguments[0].isDebug === debug
+                    )
+                    // tslint:enable: no-unsafe-any
+                }
+
+                return false
+            })
+            if (codeLenses.length === 1) {
+                return codeLenses[0]
             }
         } catch (e) {}
     }
 }
 
-async function getCodeLensesOrFail(documentUri: vscode.Uri): Promise<vscode.CodeLens[]> {
-    const codeLensPromise = getSamCodeLenses(documentUri)
-    const timeout = new Promise(resolve => {
-        setTimeout(resolve, 10000, undefined)
+async function continueDebugger(): Promise<void> {
+    await vscode.commands.executeCommand('workbench.action.debug.continue')
+}
+
+async function stopDebugger(): Promise<void> {
+    await vscode.commands.executeCommand('workbench.action.debug.stop')
+}
+
+async function closeAllEditors(): Promise<void> {
+    await vscode.commands.executeCommand('workbench.action.closeAllEditors')
+}
+
+function validateLocalInvokeResult(
+    actualResult: LocalInvokeCodeLensCommandResult,
+    expectedResult: LocalInvokeCodeLensCommandResult
+) {
+    assert.strictEqual(actualResult.datum.name, expectedResult.datum.name)
+    assert.strictEqual(actualResult.datum.value, expectedResult.datum.value)
+    assert.strictEqual(actualResult.datum.unit, expectedResult.datum.unit)
+
+    expectedResult.datum.metadata!.forEach((value, key) => {
+        assert.strictEqual(actualResult.datum.metadata!.get(key), value)
     })
-    const result = await Promise.race([codeLensPromise, timeout])
+}
 
-    assert.ok(result, 'Codelenses took too long to show up!')
+async function activateExtensions(): Promise<void> {
+    console.log('Activating extensions...')
+    // await activateExtension(VSCODE_EXTENSION_ID.python)
+    await activateExtension(VSCODE_EXTENSION_ID.awstoolkit)
+    console.log('Extensions activated')
+}
 
-    return result as vscode.CodeLens[]
+async function configureAwsToolkitExtension(): Promise<void> {
+    console.log('************************************************************')
+    // tslint:disable-next-line:no-null-keyword
```

Comment:
There's no null on this line?

Reply:
Thanks, not sure what I had that there for now, but it has been removed.